canonical/vanilla-monorepo
# feat: Typescript & Eslint configs

Adds [Typescript](https://www.typescriptlang.org/) & [Eslint](https://eslint.org/) configs.

[AirBnb](https://www.npmjs.com/package/eslint-config-airbnb) is used as a base for the Eslint configuration. 

Fixes [WD-15347](https://warthogs.atlassian.net/browse/WD-15347), [WD-15483](https://warthogs.atlassian.net/browse/WD-15483)

## QA
1. Check out this repo.
2. Install dependencies with package manager of choice: `bun install` or `yarn install` or `npm install`. 
3. CD into `apps/react_example`. 
4. Run the React app: `bun start` or `yarn start` or `npm run start`.
5. See a boilerplate React app has been compiled and launched.
6. Run linting: `bun lint` or `yarn lint` or `npm run lint`.
7. See that there are some linting errors thrown by the boilerplate test file. (Don't worry about fixing these).
8. Open `App.tsx` and introduce some formatting error. For example, you might move some attributes inside an element with multiple attributes into one line:

```tsx
        <a
          className="App-link" href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
```

9. Run linting again. See that the error is flagged by Eslint.
10. Fix the error: `bun lint-fix` or `yarn lint-fix` or `npm run lint-fix`. 
11. See that the error is fixed.

[WD-15347]: https://warthogs.atlassian.net/browse/WD-15347?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WD-15483]: https://warthogs.atlassian.net/browse/WD-15483?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
